ui/revenuecatui/src/test/kotlin/com/revenuecat/purchases/ui/revenuecatui/customercenter/data/CustomerCenterViewModelTests.kt

```diff
@@ -206,4 +194,192 @@ class CustomerCenterViewModelTests {
 
         job.join()
     }
+    
+    @Test
+    fun `dismissRestoreDialog loads customer center when restoration was successful`() = runTest {
+        // Setup basic mocks
+        every { customerInfo.activeSubscriptions } returns setOf("product-id")
+        every { customerInfo.nonSubscriptionTransactions } returns emptyList()
+        every { customerInfo.entitlements } returns EntitlementInfos(
+            emptyMap(),
+            VerificationResult.VERIFIED,
+        )
+        every { customerInfo.subscriptionsByProductIdentifier } returns emptyMap()
+
+        // Setup restore operation to return successful result
+        coEvery { purchases.awaitRestore() } returns customerInfo
+        
+        // Setup other customer center data
+        coEvery { purchases.awaitCustomerCenterConfigData() } returns configData
+        coEvery { purchases.awaitCustomerInfo(any()) } returns customerInfo
+        coEvery { purchases.awaitGetProduct(any(), any()) } returns null
+        
+        // Setup screen with a MISSING_PURCHASE path
+        val missingPurchasePath = HelpPath(
+            id = "missing_purchase_id",
+            title = "Restore Purchases",
+            type = HelpPath.PathType.MISSING_PURCHASE
+        )
+        
+        val managementScreen = Screen(
+            type = Screen.ScreenType.MANAGEMENT,
+            title = "Management",
+            subtitle = null,
+            paths = listOf(missingPurchasePath)
+        )
+        
+        every { configData.getManagementScreen() } returns managementScreen
+        
+        // Create a spy on the real ViewModel so we can verify loadCustomerCenter is called
+        val model = spyk(
```

Comment:
Not a huge fan of spies 🕵️  since I think it makes the tests do more than it should... But I think it's ok to test some end to end behaviors, like we are mostly doing here. 👍

Reply:
right... I hesitated but it was making very easy to test what I wanted

Reply:
I think I can actually remove it lol, let me check

Reply:
removed it!